pkg/tools/fake_etcd_client.go

```diff
@@ -136,7 +136,7 @@ func (f *FakeEtcdClient) Get(key string, sort, recursive bool) (*etcd.Response,
 	result := f.Data[key]
 	if result.R == nil {
 		if _, ok := f.expectNotFoundGetSet[key]; !ok {
-			f.t.Fatalf("data for %s was not defined prior to invoking Get", key)
+			f.t.Logf("data for %s was not defined prior to invoking Get", key)
```

Comment:
We are not returning random garbage, but f.NewError(EtcdErrorCodeNotFound) error. I want to handle the error in my test (I query for non-existing object), but due to f.t.Fatalf the test was dying with assertion.